src/NServiceBus.Core/Receiving/ReceiveComponent.cs

```diff
@@ -250,7 +262,7 @@ void AddReceivers(IBuilder builder, RecoverabilityExecutorFactory recoverability
             }
         }
 
-        static void LoadMessageHandlers(Configuration configuration, List<Type> orderedTypes, IConfigureComponents container, ICollection<Type> availableTypes)
+        static MessageHandlerRegistry LoadMessageHandlers(Configuration configuration, List<Type> orderedTypes, IConfigureComponents container, ICollection<Type> availableTypes)
```

Comment:
seems like configuration is only passed in to retrieve the registry from it? Why not pass the registry in as a param (this would probably also remove the return type again)?